adapters/Unity/UnityAdapter/GADMAdapterUnitySingleton.m

```diff
@@ -203,22 +180,22 @@ - (void)unityAdsDidClick:(NSString *)placementID {
 }
 
 - (void)unityAdsDidError:(UnityAdsError)error withMessage:(NSString *)message {
-  if (error == kUnityAdsErrorShowError) {
-    [_currentShowingUnityDelegate unityAdsDidError:error withMessage:message];
-    return;
-  }
-
-  NSArray *delegates;
-  @synchronized(_adapterDelegates) {
-    delegates = _adapterDelegates.objectEnumerator.allObjects;
-  }
-
-  for (id<UnityAdsExtendedDelegate, UnityAdsExtendedDelegate> delegate in delegates) {
-    [delegate unityAdsDidError:error withMessage:message];
-  }
+  if (error == kUnityAdsErrorInitSanityCheckFail || error == kUnityAdsErrorNotInitialized || error == kUnityAdsErrorInvalidArgument ||
```

Comment:
Will the UnityAds SDK invoke unityAdsDidError:withMessage: for each call to [UnityAds initialize:testMode:enablePlacementPerLoad:]? I ask because, once we get this failure once, any future ad loads will still see initialized to false and try to initialize again. I want to make sure we're waiting on a callback that will actually come, and not just time out.

If we can't expect multiple callbacks, the adapter should keep track of its own initialization state to track this.

Please confirm, this is essentially my only blocking comment.

Reply:
Yes, any failed initialization will trigger these errors

Reply:
Thanks for confirming!